share/osutil/file_linux.go

```diff
@@ -95,21 +134,26 @@ func GetContainerRealFilePath(pid int, symlinkPath string) (string, error) {
 		for i := range parts {
 			partialSymlink := strings.Join(parts[i:], "/")
 			resolvedPath = filepath.Join(filepath.Dir(currentPath), partialSymlink)
-			if strings.HasPrefix(resolvedPath, procRoot) && fileExists(resolvedPath) {
+			// Assume the first resolved path under proc root is the correct path
+			if strings.HasPrefix(resolvedPath, procRoot) {
+				underProcRoot = true
 				break
 			}
 		}
 
-		// nest link
-		if strings.HasPrefix(resolvedPath, procRoot) && fileExists(resolvedPath) {
+		if underProcRoot {
+			if err = fileExists(resolvedPath); err != nil {
+				log.WithFields(log.Fields{"resolvedPath": resolvedPath}).Debug("File not exist")
+				return "", err
+			}
+
+			// nest link
 			if symlink, err = os.Readlink(resolvedPath); err == nil {
 				visitedSymlink[currentPath] = struct{}{}
 				currentPath = resolvedPath;
 			} else {
 				return resolvedPath, nil 
 			}
-		} else {
-			break
 		}
```

Comment:
`fileExists()` and `os.Readlink()` are called twice in one single round of this loop.  Any chance to not call them twice?

Note: caller already knows symlinkPath is a symlink and it exists.